core/chains/evm/client/rpc_client.go

```diff
@@ -0,0 +1,945 @@
+	if err == nil {
+		r.registerSub(sub)
+	}
+	err = r.wrapWS(err)
+	duration := time.Since(start)
+
+	r.logResult(lggr, err, duration, r.getRPCDomain(), "SubscribeFilterLogs")
+
+	return
+}
+
+func (r *rpcClient) SuggestGasTipCap(ctx context.Context) (tipCap *big.Int, err error) {
+	ctx, cancel, ws, http, err := r.makeLiveQueryCtxAndSafeGetClients(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer cancel()
+	lggr := r.newRqLggr()
+
+	lggr.Debug("RPC call: evmclient.Client#SuggestGasTipCap")
+	start := time.Now()
+	if http != nil {
+		tipCap, err = http.geth.SuggestGasTipCap(ctx)
+		err = r.wrapHTTP(err)
+	} else {
+		tipCap, err = ws.geth.SuggestGasTipCap(ctx)
+		err = r.wrapWS(err)
+	}
+	duration := time.Since(start)
+
+	r.logResult(lggr, err, duration, r.getRPCDomain(), "SuggestGasTipCap",
+		"tipCap", tipCap,
+	)
+
+	return
+}
+
+func (r *rpcClient) ChainID() (chainID *big.Int, err error) { return r.chainID, nil }
+
+// Returns the ChainID according to the geth client. This is useful for functions like verify()
+// the common node.
+func (r *rpcClient) ClientChainID(ctx context.Context) (chainID *big.Int, err error) {
+	ctx, cancel, ws, http, err := r.makeLiveQueryCtxAndSafeGetClients(ctx)
+
+	defer cancel()
+
+	if http != nil {
+		chainID, err = http.geth.ChainID(ctx)
+		err = r.wrapHTTP(err)
+	} else {
+		chainID, err = ws.geth.ChainID(ctx)
+		err = r.wrapWS(err)
+	}
+	return
+}
+
+func (r *rpcClient) ConfiguredChainID() *big.Int {
+	chainID, _ := r.ChainID()
+	return chainID
+}
+
+// newRqLggr generates a new logger with a unique request ID
+func (r *rpcClient) newRqLggr() logger.Logger {
+	return r.rpcLog.With(
+		"requestID", uuid.New(),
+	)
+}
+
+func (r *rpcClient) wrapWS(err error) error {
+	err = wrap(err, fmt.Sprintf("primary websocket (%s)", r.ws.uri.Redacted()))
+	return err
+}
+
+func (r *rpcClient) wrapHTTP(err error) error {
+	err = wrap(err, fmt.Sprintf("primary http (%s)", r.http.uri.Redacted()))
+	if err != nil {
+		r.rpcLog.Debugw("Call failed", "err", err)
+	} else {
+		r.rpcLog.Trace("Call succeeded")
+	}
+	return err
+}
+
+// makeLiveQueryCtxAndSafeGetClients wraps makeQueryCtx but returns error if rpcClient is not clienttypes.NodeStateAlive.
+func (r *rpcClient) makeLiveQueryCtxAndSafeGetClients(parentCtx context.Context) (ctx context.Context, cancel context.CancelFunc, ws rawclient, http *rawclient, err error) {
+	// Need to wrap in mutex because state transition can cancel and replace the
+	// context
+	r.stateMu.RLock()
+	if r.state != clienttypes.NodeStateAlive {
+		err = errors.Errorf("cannot execute RPC call on rpcClient with state: %s", r.state)
+		r.stateMu.RUnlock()
+		return
+	}
+	cancelCh := r.chStopInFlight
+	ws = r.ws
+	if r.http != nil {
+		cp := *r.http
+		http = &cp
+	}
+	r.stateMu.RUnlock()
+	ctx, cancel = makeQueryCtx(parentCtx, cancelCh)
+	return
+}
+
+func (r *rpcClient) makeQueryCtx(ctx context.Context) (context.Context, context.CancelFunc) {
+	return makeQueryCtx(ctx, r.getChStopInflight())
```

Comment:
This seems to call the `makeQueryCtx` method from the evm node we are trying to deprecate. Is this correct?